src/common/statbar.cpp

```diff
@@ -297,4 +299,50 @@ void wxStatusBarBase::PopStatusText(int number)
         DoUpdateStatusText(number);
 }
 
+// ----------------------------------------------------------------------------
+// controls
+// ----------------------------------------------------------------------------
+bool wxStatusBarBase::AddFieldControl(int n, wxWindow* win)
+{
+    wxCHECK_MSG( (unsigned)n < m_panes.size(), false,
+                    "invalid status bar field index" );
+    wxCHECK_MSG( !m_panes[n].IsFieldControl(), false,
+                    "another control is already added in this field" );
+
+    m_panes[n].SetFieldControl(win);
+
+    return true;
+}
+
+void wxStatusBarBase::OnSize(wxSizeEvent& event)
+{
+    event.Skip();
+
+    auto GetFieldNumber = [this](wxWindow* win)
```

Comment:
If we stored windows instead of IDs we could simplify this by just iterating over the panes, checking if each pane has a window and repositioning the window if so.